editor/src/components/inspector/controls/advanced-grid-modal.tsx

```diff
@@ -0,0 +1,251 @@
+import React from 'react'
+import { InspectorModal } from '../widgets/inspector-modal'
+import { FlexColumn, H1, SquareButton, UtopiaStyles } from '../../../uuiui'
+import { UIGridRow } from '../widgets/ui-grid-row'
+import type { OptionChainOption } from './option-chain-control'
+import { OptionChainControl } from './option-chain-control'
+import { PrettyLabel } from '../sections/layout-section/flex-container-subsection/flex-container-controls'
+import type { InspectorInfo } from '../common/property-path-hooks'
+import { useInspectorLayoutInfo } from '../common/property-path-hooks'
+import {
+  RadixSelect,
+  regularRadixSelectOption,
+  separatorRadixSelectOption,
+} from '../../../uuiui/radix-components'
+import { optionalMap } from '../../../core/shared/optional-utils'
+import { FlexJustifyContent } from 'utopia-api/core'
+
+export interface AdvancedGridModalProps {
+  id: string
+  testId: string
+  key: string
+  popupOpen?: boolean
+  openPopup?: (id: string) => void
+  closePopup?: () => void
+  style?: React.CSSProperties
+  pickerOffset?: {
+    x: number
+    y: number
+  }
+}
+
+export const AdvancedGridModal = React.memo((props: AdvancedGridModalProps) => {
+  const pickerOffset = props.pickerOffset ?? { x: -280, y: -20 }
```

Comment:
this feels very hacky, can't we anchor the picker properly so it doesn't need to know about a magic number which is approximately the inspector width?

Reply:
definitely, this was already in the original code so I missed it - I'll fix

Reply:
automatically getting the offset from the panel width now - and just setting the desired distance from it